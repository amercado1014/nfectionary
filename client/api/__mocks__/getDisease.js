import * as mock from '../../mockData/mockData';

export const getDisease = jest.fn().mockImplementation(() => {
  return Promise.resolve(mock.diseaseObject);
});
