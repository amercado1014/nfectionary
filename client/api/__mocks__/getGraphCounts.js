import * as mock from '../../mockData/mockData';

export const getGraphCounts = jest.fn().mockImplementation(() => {
  return Promise.resolve(mock.returnedGraphCounts);
});
