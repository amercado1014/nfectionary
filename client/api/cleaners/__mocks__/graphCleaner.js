import * as mock from '../../../mockData/mockData';

export const graphCleaner = jest.fn().mockImplementation(() => {
  return mock.returnedGraphData;
});
