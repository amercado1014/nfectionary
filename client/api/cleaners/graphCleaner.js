export const graphCleaner = (rawData, shortNames) => {

  const graphData = shortNames.reduce((accu, state) => {

    rawData.filter(num => {
      
      let newObject = {};
      if (num.state === state.id) {
        let string = "No Cases Reported";
        let countNum = num.count !== string ? num.count : '0';
        newObject['state'] = state.state;
        newObject['count'] = parseInt(countNum);
      }
      if (newObject.state) {
        return accu.push(newObject);
      }
    });
    return accu;
  }, []);
  return graphData;
};


