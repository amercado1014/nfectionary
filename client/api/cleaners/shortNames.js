export const shortNames = [
  { state: 'AL', id: 1 },
  { state: 'AK', id: 2 },
  { state: 'AZ', id: 3 },
  { state: 'AR', id: 4 },
  { state: 'CA', id: 5 },
  { state: 'CO', id: 6 },
  { state: 'CT', id: 7 },
  { state: 'DE', id: 8 },
  { state: 'FL', id: 9 },
  { state: 'GA', id: 10 },
  { state: 'GU', id: 11 },
  { state: 'HI', id: 12 },
  { state: 'ID', id: 13 },
  { state: 'IL', id: 14 },
  { state: 'IN', id: 15 },
  { state: 'IA', id: 16 },
  { state: 'KS', id: 17 },
  { state: 'KY', id: 18 },
  { state: 'LA', id: 19 },
  { state: 'ME', id: 20 },
  { state: 'MD', id: 21 },
  { state: 'MA', id: 22 },
  { state: 'MI', id: 23 },
  { state: 'MN', id: 24 },
  { state: 'MS', id: 25 },
  { state: 'MO', id: 26 },
  { state: 'MT', id: 27 },
  { state: 'NE', id: 28 },
  { state: 'NV', id: 29 },
  { state: 'NH', id: 30 },
  { state: 'NJ', id: 31 },
  { state: 'NM', id: 32 },
  { state: 'NY', id: 33 },
  { state: 'NC', id: 34 },
  { state: 'ND', id: 35 },
  { state: 'OH', id: 36 },
  { state: 'OK', id: 37 },
  { state: 'OR', id: 38 },
  { state: 'PA', id: 39 },
  { state: 'PR', id: 40 },
  { state: 'RI', id: 41 },
  { state: 'SC', id: 42 },
  { state: 'SD', id: 43 },
  { state: 'TN', id: 44 },
  { state: 'TX', id: 45 },
  { state: 'UT', id: 46 },
  { state: 'VT', id: 47 },
  { state: 'VA', id: 48 },
  { state: 'WA', id: 49 },
  { state: 'WV', id: 50 },
  { state: 'WI', id: 51 },
  { state: 'WY', id: 52 }
];
