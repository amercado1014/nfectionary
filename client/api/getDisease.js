export const getDisease = async id => {
  try {
    const response = await fetch(
      `http://localhost:5000/api/v1/diseases/${id}`
    );
    const disease = await response.json();
    return disease;
  } catch (err) {
    throw new Error(`Error fetching disease information: ${err.message}`);
  }
};
