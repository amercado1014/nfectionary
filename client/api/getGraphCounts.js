export const getGraphCounts = async (id) => {
  try {
    const response = await fetch(`http://localhost:5000/api/v1/specific-disease/${id}`);
    
    const diseases = await response.json();
    
    const diseaseCount = diseases.map(disease => ({ 
      count: disease.case_count, 
      id: disease.diseases_id, 
      state: disease.states_id
    }));
    return diseaseCount
  } catch(err) {
    throw new Error(err.message);
  }
};