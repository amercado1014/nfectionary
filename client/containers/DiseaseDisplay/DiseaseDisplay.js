import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Image,
  ScrollView
} from 'react-native';
import { getDisease } from '../../api/getDisease';
import { graphCleaner } from '../../api/cleaners/graphCleaner';
import { shortNames } from '../../api/cleaners/shortNames';
import {
  VictoryBar,
  VictoryChart,
  VictoryTheme,
  createContainer
} from "victory-native";
import { getGraphCounts } from '../../api/getGraphCounts';
import Accordion from 'react-native-collapsible/Accordion';
import ModalDropdown from 'react-native-modal-dropdown';
import PropTypes from 'prop-types';

export default class DiseaseDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      diseaseInfo: {},
      graphData: [],
      diseaseList: [],
      activeSection: false,
      collapsed: true
    };
  }

  async componentDidMount() {
    const id = this.props.navigation.getParam('disease_id');

    await this.fetchDiseaseData(Number(id) - 1);
  }

  async fetchDiseaseData(id) {
    const diseaseId = Number(id) + 1;
    const disease = await getDisease(diseaseId);
    const rawData = await getGraphCounts(diseaseId);
    const graphData = graphCleaner(rawData, shortNames);
    const diseaseList = this.props.navigation.getParam('diseaseList')
      .map(disease => disease.name);

    this.setDiseaseInfo(disease, graphData, diseaseList);
  }

  setDiseaseInfo(diseaseInfo, graphData, diseaseList) {
    this.setState({
      diseaseInfo,
      graphData,
      diseaseList
    });
  }

  _toggleExpanded() {
    this.setState({
      collapsed: !this.state.collapsed
    });
  }

  _setSection(section) {
    this.setState({
      activeSection: section
    });
  }

  _renderHeader(section, i, isActive) {
    return ( 
      <View style = {[
        styles.header, 
        isActive ? styles.active : styles.inactive
      ]}>
        <Text style = {styles.headerText}> { section.title }</Text> 
      </View>
    );
  }

  _renderContent(section, i, isActive) {
    return ( 
      <Text style = {[
        styles.content, 
        isActive ? styles.active : styles.inactive
      ]}> 
        {section.content} 
      </Text>
    );
  }

  // eslint-disable-next-line
  static navigationOptions = {
    title: ' ',
    headerStyle: {
      backgroundColor: '#3E79CA',
      borderBottomColor: '#3E79CA'
    },
    headerTintColor: '#fff',
    headerTitleStyle: {
      color: '#ffffff',
      fontSize: 30,
      textDecorationLine: 'underline'
    }
  };

  render() {
    const {
      name,
      treatment,
      preventative_measures,
      signs_symptoms,
      images,
      testing_procedures,
      transmission,
      summary
    } = this.state.diseaseInfo;
    const { diseaseList } = this.state;

    const diseaseContent = [
      { title: 'Description', content: summary },
      { title: 'Treatment:', content: treatment },
      { title: 'Preventative Measures', content: preventative_measures },
      { title: 'Transmission', content: transmission },
      { title: 'Testing Procedures', content: testing_procedures },
      { title: 'Signs/ Symptoms', content: signs_symptoms }
    ];
  
    const VictoryZoomVoronoiContainer = createContainer("zoom", "voronoi");

    return ( 
      <View style={{flex: 1}}>
      <ScrollView contentContainerStyle = {styles.container}>
        <View style = {styles.dropDownWrap}>
          <ModalDropdown 
            defaultValue={name}
            style={styles.dropDown}
            options={diseaseList} 
            textStyle={{ color: '#3E79CA', fontSize: 20, textAlign: 'left', width: 160 }}
            dropdownTextStyle={ {color: '#3E79CA', fontSize: 16} }
            onSelect = {(event) => this.fetchDiseaseData(event)}
          />
          <View style={styles.arrWrap}><Text style={styles.arrColor}>â–¼</Text></View>
        </View>
      <Text style={styles.chartTitle}> Reported Cases - YTD </Text> 
      { this.state.graphData.length > 0 ?
          <VictoryChart
            height={350}
            style = {styles.chart}
            containerComponent = {
                <VictoryZoomVoronoiContainer 
                  voronoiDimension = "x"
                  labels = {(d) => `cases: ${d.y}`}
                  zoomDomain = {{x: [5, 15]}}
                  allowZoom = {false}
                />
              }
          >
            <VictoryBar
              style = {{data: {fill: "#c43a31", stroke: "black", strokeWidth: 1 }}}
              alignment = "start"
              barRatio = {2.5}
              data = {this.state.graphData}
              x = "state"
              y = "count" 
            />
            </VictoryChart> : <Text> Loading Chart Data </Text>
          } 
        <Image
          resizeMode="contain"
          style = { styles.image }
          source = { {url: `${images}`} }>
        </Image> 

        <Accordion
          activeSection = { this.state.activeSection }
          sections = { diseaseContent }
          touchableComponent = { TouchableOpacity }
          renderHeader = { this._renderHeader }
          renderContent = { this._renderContent }
          onChange = { this._setSection.bind(this) }
        /> 
       </ScrollView>
       </View>
      );
    }
  }

  
  const styles = StyleSheet.create({
    container: {
      backgroundColor: '#3E79CA',
      alignItems: 'center',
      justifyContent: 'space-between',
      paddingTop: 20,
      paddingBottom: 20
    },
    dropDownWrap: {
      alignItems:'center', 
      flexDirection:'row'
    },
    dropDown: {
      backgroundColor: '#ffffff',
      marginBottom: 20,
      padding: 10,
      width: 190,
      borderRadius: 3,
      shadowOffset:{  width: 1,  height: 1,  },
      shadowColor: 'black',
      shadowOpacity: .5
    },
    arrWrap: {
      position: "absolute", 
      right: 15, 
      top: 15
    },
    arrColor: {
      color: '#3E79CA'
    },
    chart: {
      color: '#ffffff',
      height: 320,
      width: 320,
      top: 0,
      left: 0,
      bottom: 0,
      right: 0,
    },
    chartTitle: {
      color: '#ffffff',
      fontSize: 20,
      textAlign: 'center'
    },
    image: {
      position: 'relative',
      top: 0,
      left: 0,
      bottom: 0,
      right: 0,
    },
    textBlock: {
      color: '#ffffff',
      marginTop: 10,
      marginBottom: 10
    },
    title: {
      textAlign: 'left',
      fontSize: 22,
      fontWeight: '300',
      marginBottom: 20
    },
    header: {
      backgroundColor: '#F5FCFF',
      padding: 10
    },
    headerText: {
      textAlign: 'left',
      fontSize: 16,
      fontWeight: '500'
    },
    content: {
      padding: 20,
      backgroundColor: 'rgba(255,255,255,1)'
    },
    active: {
      backgroundColor: 'rgba(255,255,255,1)'
    },
    inactive: {
      backgroundColor: 'rgba(245,252,255,1)'
    }
  });
