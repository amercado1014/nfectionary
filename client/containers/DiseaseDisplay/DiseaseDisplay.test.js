/* eslint-disable max-len */
import React from 'react';
import DiseaseDisplay from './DiseaseDisplay';
import Enzyme, { shallow } from 'enzyme';
import { getDisease } from '../../api/getDisease';
import { getGraphCounts } from '../../api/getGraphCounts';
import { shortNames } from '../../api/cleaners/shortNames';
import { graphCleaner } from '../../api/cleaners/graphCleaner';
import * as mock from '../../mockData/mockData';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

jest.mock('../../api/getDisease');
jest.mock('../../api/getGraphCounts');
jest.mock('../../api/cleaners/graphCleaner');

describe('DiseaseDisplay', () => {
  let wrapper;
  let instance;
  let id;
  let disease;
  let diseaseList;
  let graphData;

  beforeEach(() => {
    wrapper = shallow(<DiseaseDisplay navigation={{ getParam: jest.fn().mockImplementation(()=> {
      return mock.diseaseNames; 
    }) 
    }} />);
    instance = wrapper.instance();
    id = '1';
    disease = mock.diseaseObject;
    diseaseList = mock.diseases;
    graphData = mock.returnedGraphData;
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
      diseaseInfo: {},
      graphData: [],
      diseaseList: [],
      activeSection: false,
      collapsed: true
    });
  });

  it('should call fetchDiseaseData on componentDidMount', async () => {
    instance.fetchDiseaseData = jest.fn();
    await instance.componentDidMount();
    expect(instance.fetchDiseaseData).toHaveBeenCalled();
  });

  it('should call getDisease on fetchDiseaseData with correct params', async () => {
    const expected = 2;
    await instance.fetchDiseaseData(id);
    expect(getDisease).toHaveBeenCalledWith(expected);
  });

  it('should call getGraphCounts on fetchDiseaseData with correct params', async () => {
    const expected = 2;
    await instance.fetchDiseaseData(id);
    expect(getGraphCounts).toHaveBeenCalledWith(expected);
  });

  it('should call graphCleaner on fetchAllData with correct params', async () => {
    const rawData = mock.returnedGraphCounts;
    await instance.fetchDiseaseData(id);
    expect(graphCleaner).toHaveBeenCalledWith(rawData, shortNames);
  });

  it('should call setDiseaseInfo on fetchAllData with correct params', async () => {
    instance.setDiseaseInfo = jest.fn();
    await instance.fetchDiseaseData(id);
    expect(instance.setDiseaseInfo).toHaveBeenCalledWith(disease, graphData, diseaseList);
  });

  it('should set state with diseaseInfo on setDiseaseInfo', () => {
    instance.setDiseaseInfo(disease, graphData, diseaseList);
    expect(wrapper.state('diseaseInfo')).toEqual(disease);
  });

  it('should set state with graphData on setDiseaseInfo', () => {
    instance.setDiseaseInfo(disease, graphData, diseaseList);
    expect(wrapper.state('graphData')).toEqual(graphData);
  });

  it('should set state with diseaseList on setDiseaseInfo', () => {
    instance.setDiseaseInfo(disease, graphData, diseaseList);
    expect(wrapper.state('diseaseList')).toEqual(diseaseList);
  });
});
