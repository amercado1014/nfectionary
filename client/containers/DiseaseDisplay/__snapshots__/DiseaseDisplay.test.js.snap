// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <Image
    source={
      Object {
        "url": "undefined",
      }
    }
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Treatment: 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Preventative Measures: 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Transmission: 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Testing Procedures: 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Signs/ Symptoms: 
  </Text>
</View>
`;
